src/js/state/Viewer/reducer.js

```diff
@@ -1,25 +1,31 @@
 /* @flow */
+import produce from "immer"
 
 import type {ViewerAction, ViewerState} from "./types"
 import {concat, splice} from "../../lib/Array"
+import {createSelection} from "./helpers/selection"
 
-const init: ViewerState = {
+const init = (): ViewerState => ({
   records: [],
   endStatus: "INCOMPLETE",
   status: "INIT",
   columns: {},
-  scrollPos: {x: 0, y: 0}
-}
+  scrollPos: {x: 0, y: 0},
+  selection: {
+    rows: {},
+    currentRange: [0, 0]
+  }
+})
 
 export default function(
-  state: ViewerState = init,
+  state: ViewerState = init(),
```

Comment:
👍 In addition to being careful about ensuring initial state does not reuse a pointer to an old object, the changes here are reminding me that we should prob choose to continue to adopt `immer` with the same policy we've decided for `styled-components`

Reply:
Yes, that makes sense to me.